src/endpoint/s3/ops/s3_get_bucket_policy_status.js

```diff
@@ -16,27 +17,17 @@ async function get_bucket_policy_status(req) {
 // TODO: implemented according to current implementation of authorize_request_policy. should update when authorize_request_policy changed
 // full public policy defintion: https://docs.aws.amazon.com/AmazonS3/latest/dev/access-control-block-public-access.html#access-control-block-public-access-policy-status
 function _is_policy_public(policy) {
-    for (const statement of policy.statement) {
-        let principal_wildcard = false;
-        let resource_wildcard = false;
-        if (statement.effect === 'deny') {
-            return false;
-        }
-        for (const principal of statement.principal) {
-            if (principal.unwrap() === '*') {
-                principal_wildcard = true;
-            }
-        }
-        for (const resource of statement.resource) {
```

Comment:
is this something you checked with AWS? because in all the examples they have the resource is *

Reply:
yes, I checked it. when adding a policy with specific principles and resources with a wildcard it returns 'false' for this function. it did return true for specific resource and wildcard principals. Also looking at the definition of public it doesn't seem to list resources as one of the things evaluated for public policy. I am not sure why in their examples the resource is always *